src/control/threadlocalquickaction.h

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+#include <QObject>
+#include <QThreadStorage>
+#include <memory>
+#include <utility>
+
+#include "control/controlpushbutton.h"
+#include "preferences/configobject.h"
+
+class QuickAction;
+class ControlPushButton;
+
+// A class that lazily creates a QuickAction instance for each thread.
+//
+// Public methods are thread safe
+class ThreadLocalQuickAction : public QObject {
```

Comment:
What's the purpose of deriving from QObject if the class is not to be used within an event loop?

Reply:
I need it to be a QObject because of the slot.

Reply:
`void slotTriggered(double);` is not marked as a slot

Reply:
Ok, I see that with the new signals and slots syntax slotTriggered does not need to be marked as slot. I've removed QObject from here.

Reply:
Actually I can't. the function does not need to be marked as slot, but the object still needs to be a QObject to use connect.

Reply:
I marked the function as a slot for clarity.